### DISCUSSION TITLE
Why an error after changing the sort order of underlying data?

### DISCUSSION THREAD

**Discussion by andrewmrichards** [team-moeller/better-access-charts#141] [Q&A] [answered]
Hi
I saw your presentation at the Europe AUG, and loved what you've done - thanks so much.
I'm now trying to use the tool for a client database.
The form using the chart allows the user to choose to see the top X people, or the bottom X people in the chart. When they've made the choice, the only difference in the code is reflected in these lines:

````
    If Left(Me.cboChart, 3) = "Top" Then
        strSQL = "SELECT * FROM WorkloadHistory ORDER BY WorkloadPoints DESC"
    Else
        strSQL = "SELECT * FROM WorkloadHistory ORDER BY WorkloadPoints ASC"
    End If
````

If the user chooses to see the top X items, the chart works perfectly, but if they choose the bottom X items, I get an error
Expected ']'.

The line generating this error is in the BAC_Chart module, within CreateChart, on the execution of the line
````
    .Write strHtml
````
I've attached samples of the two HTML files created - the one which works and the one which doesn't. I've also attached a file containing the data which should be the source of the chart - I wondered if there were "weird" characters in someone's name or something like that, but I can't see anything

I know virtually nothing about js, and little about HTML, but I can't see any reason why the error would occur based simply on the data being sorted differently, so am I likely to be missing something else?

Thanks for your help.

**Comment by team-moeller** [comment]
Hi @andrewmrichards !
Where can I find the attached files? I do not see them.

**Comment by andrewmrichards** [comment]
Hi Thomas

I uploaded them to the Github comments / report.

But here they are attached to this email too.

Many thanks
Andrew

**Comment by andrewmrichards** [comment]
Hi again

I’ve now taken a closer look at this in order to try to replicate the issue.

The problem is in one row of data, as highlighted here:

Postholder
Category
WorkloadPoints
Andrea Brazzoduro
Examining
3
INE JACOBS
Examining
65.5
Judith Brown
Examining
15
Kevin O'Rourke
Examining
69
Leonardo Davoudi
Examining
42
Lucy Clarke
Examining
17

The problem is with the data for Kevin O’Rourke – the aspostrophe causes the script to break. Take out the apostrophe and it works fine.

I’ve tried replacing the apostrophe with a double-apostrophe but this didn’t help…

I can adjust the code which populates this temp table with a tweak to remove any apostrophes, which is fine for now, but if there’s any way of adjusting the script to allow for apostrophes in names, that would be great!

Thanks for your help.

Andrew

From: Andrew Richards
Sent: Friday, July 21, 2023 11:52 AM
To: team-moeller/better-access-charts ***@***.***>
Subject: RE: [team-moeller/better-access-charts] Why an error after changing the sort order of underlying data? (Discussion #141)

Hi Thomas

I uploaded them to the Github comments / report.

But here they are attached to this email too.

Many thanks
Andrew

**Comment by team-moeller** [comment, marked as answer]
Hi @andrewmrichards ,
it has taken a long time but today I found a solution for your problem.
Please try version [4.02.01](https://github.com/team-moeller/better-access-charts/releases/tag/4.02.01)
Please consider to close this discussion, when the new version works for you.
Regards!
Thomas
